### PR TITLE
Added a check for the webserver port availability

### PR DESCRIPTION
**Pull request by miniche**
A function in the new "web" module checks if a port is used (on the local or a remote computer).

The test have been added when CentralReport is starting. In this case, the webserver is not loaded and an error is written in logs.
This solves the issue #65.

### CODE REVIEW

**Comment by miniche** [comment]
Ready to be merged

File: NEWS [hunk 1]
```diff
@@ -17,6 +17,10 @@ Version Alpha 0.3.0 (Not released yet)
           All confirmations are validated by default.
           Useful to perform tests or for deployments.
 
+    - Improvement #72: Added a check for the webserver port availability
+        * When the choosen port is already used on the current computer,
```
Comment (Ninir) [inline NEWS]: Typo: `Chosen`

`When the chosen port is already in use on the current host, the webserver isn't started and an error is logged.`

File: centralreport/centralreport.py [hunk 2]
```diff
@@ -93,12 +94,18 @@ def run(self):
...
-        # Is webserver enabled?
+        # Launching the internal webserver...
         if not is_error and text.convert_text_to_bool(Config.get_config_value('Webserver', 'enable')):
-            from web.server import WebServer
+            local_web_port = int(Config.get_config_value('Webserver', 'port'))
 
-            log.log_info('Enabling the webserver...')
-            CentralReport.webserver_thread = WebServer()
+            if not utils_web.check_port('127.0.0.1', local_web_port):
+                log.log_info('Enabling the webserver...')
```
Comment (Ninir) [inline centralreport/centralreport.py]: Are we really `enabling` the web server here or are we starting it?

File: centralreport/centralreport.py [hunk 3]
```diff
@@ -93,12 +94,18 @@ def run(self):
...
+
+                # Importing the module here improve the memory usage
+                from web.server import WebServer
+                CentralReport.webserver_thread = WebServer()
+            else:
+                log.log_error('Error launching the webserver: port %s is already used on this host!' % local_web_port)
```
Comment (Ninir) [inline centralreport/centralreport.py]: I would write `in use` instead of used

File: centralreport/cr/utils/web.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# -*- coding: utf-8 -*-
+
+"""
+    CentralReport - Web module
+        Contains useful functions around the web
```
Comment (Ninir) [inline centralreport/cr/utils/web.py]: AFAIR, it is `from around`

File: centralreport/cr/utils/web.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    https://github.com/CentralReport
+"""
+
+import socket
+
+
+def check_port(ip, port):
+    """
+        Checks if the port is open on a specific IP
```
Comment (Ninir) [inline centralreport/cr/utils/web.py:15]: `is opened`

**Comment by Ninir** [comment]
Seems good

File: centralreport/centralreport.py (hunk 2 above)
Comment (miniche) [inline centralreport/centralreport.py]: In this case, the verb "to enable" is a synonym of "to activate", isn't it?
We are not just starting the web server: we are loading the python module and all the libraries required, initialize CherryPy and after, we are starting the web server.

I think it's another a detail. What is the best word, in your opinion?

Comment (Ninir) [inline centralreport/centralreport.py]: Initialize fits better our needs then IMO

File: centralreport/cr/utils/web.py (hunk 4 above)
Comment (miniche) [inline centralreport/cr/utils/web.py]: True, thanks!

File: centralreport/cr/utils/web.py (hunk 5 above)
Comment (miniche) [inline centralreport/cr/utils/web.py:15]: Are you sure?
http://www.usingenglish.com/forum/ask-teacher/15771-open-opened-welcome-welcomed.html#post73496

**Comment by Guimove** [comment]
:+1:

**Comment by miniche** [comment]
Tested with Debian 6 and Ubuntu 12.04. Will be merged in the next seconds :)

**Pull request merged**